packages/studio-base/src/players/FoxgloveDataPlatformPlayer/index.ts

```diff
@@ -168,13 +168,14 @@ export default class FoxgloveDataPlatformPlayer implements Player {
     if (isLessThan(this._start, coverageStart)) {
       log.debug("Reduced start time from", this._start, "to", coverageStart);
       this._start = coverageStart;
-      this._currentTime = this._start;
     }
     if (isGreaterThan(this._end, coverageEnd)) {
       log.debug("Reduced end time from", this._end, "to", coverageEnd);
       this._end = coverageEnd;
     }
 
+    this._currentTime = clampTime(this._currentTime, this._start, this._end);
```

Comment:
This is an initialization function; the currentTime was just set in the constructor. With my current reading of the code I don't think there is any other change to currentTime since the constructor so there's no need for clamping logic here.

```suggestion
    this._currentTime = this._start;
```

Reply:
This code is putting back the bug I am fixing. `open` is async. During startup, the `urlInitialState` hook in the URLStateAdapter will call seekPlayback to set the currentTime to the time the user specified in the URL. If the start/end range need to clamp, the logic of setting currentTime to the start time overrides this.

I need to add a comment because changing the code back to what you have is _not_ the behavior we want.